app/components/provider_interface/activity_log_event_component.rb

```diff
@@ -45,14 +45,23 @@ def event_description
       else
```

Comment:
This case doesn't cover the `conditions_not_met` status, so we get blank titles in those blocks (there are a couple of examples in the review app). Is there a nice way to ensure that the specs cover all cases for the status and that were returning a sensible answer for each one?

Reply:
In theory, it'd be possible to write something based on `ApplicationStateChange.states_visible_to_provider`, but we have too many variations, e.g. multiple types of rejection, declining etc. so this remains a manual check for now.